client/actions/post.js

```diff
@@ -0,0 +1,69 @@
+// Action Types
+const LOADPOST = 'LOADPOST';
+const LOADPOSTS = 'LOADPOSTS';
+const FAILEDTOLOADPOST = 'FAILEDTOLOADPOST';
+const REMOVEPOST = 'REMOVEPOST';
+const FAILEDTODELETEPOST = 'FAILEDTODELETEPOST';
+
+// Actions
+export const _loadPost = (post) => ({
+    type: LOADPOST,
+    post,
+});
+
+export const _loadPosts = (posts) => ({
+    type: LOADPOSTS,
+    posts
+})
+
+const _failedtoDeleteUser = () => ({ type: FAILEDTODELETEPOST });
+
+const _removePost = () => ({ type: REMOVEPOST });
+
+const _failedToLoadPost = () => ({ type: FAILEDTOLOADPOST });
+
+// Thunks
+export const loadPost = (postId) => async (dispatch, _, axios) => {
+    try {
+        const { data: post } = await axios.get(`/api/posts/${postId}`);
+        if (post) {
+            dispatch(_loadPost(post));
+        } else {
+            dispatch(_failedToLoadPost());
+        }
+    } catch (err) {
+        dispatch(_failedToLoadPost());
+        console.error(err);
+    }
+};
+
+export const removePost = (postId) => async (dispatch, _, axios) => {
+    try {
+        const { data: deleted } = await axios.delete(`/api/post/deletePost/${postId}`);
+        if (deleted) {
+            dispatch(_removePost());
+        } else {
+            dispatch(_failedtoDeleteUser())
+        }
+    } catch (err) {
+        dispatch(_failedtoDeleteUser());
+        console.error(err);
+    }
+};
+
+export const loadPosts = (id, type) => async (dispatch, _, axios) => {
+    switch (type) {
+        case 'group': {
+            const posts = await axios.get(`/api/posts/groupPosts/${id}`);
+            dispatch(_loadPosts(posts));
+        }
+            break;
```

Comment:
Pretty sure break; needs to go inside the braces for the first case.

Reply:
Oddly enough `break` doesn't need to be in the curly braces for the specific case, but I agree its good form to do that. I recommend the same @CB6670.

Having a break statement outside the case will still get executed based on the way switches *fall through*. This could have the effect of breaking out of the switch when you didn't intend to though. So try to be mindful of that. Switches are a bit of an odd animal.

Reply:
Ok, thanks for letting me know.